src/ast_transform.coffee

```diff
@@ -604,7 +590,26 @@ do ()=>
               arg.t.name = _case.var_decl.name
               arg.t.type = _case.var_decl.type
               arg.name = arg_name
+
+            if func.name == "constructor"
```

Comment:
Should check if "constructor" is included in name because of `v.name = "#{parent.name}_constructor"` at https://github.com/madfish-solutions/sol2ligo/pull/148/files#diff-1dac19895cc2e00fee8214d6f6c1ec72R742